### PR TITLE
Csp 358 filter out onboarding providers from endpoints

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.Roatp.Data/Repositories/ProviderCoursesReadRepository.cs [hunk 1]
```diff
@@ -45,13 +47,16 @@ public async Task<List<ProviderCourse>> GetAllProviderCourses(int ukprn)
...
 
-        public async Task<int> GetProvidersCount(int larscode)
+        public Task<int> GetProvidersCount(int larsCode)
         {
-            return await _roatpDataContext
-                .ProviderCourses
-                .AsNoTracking()
-                .Where(c => c.LarsCode == larscode)
-                .CountAsync();
+            FormattableString sql = $@"SELECT DISTINCT PC.*
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.Data/Repositories/ProviderCoursesReadRepository.cs:52]: There was no way I could find to run the sql directly, I could only map it to the ProviderCourse entity then count it.  All the examples I could find online said to use 'FromSQL' and map to an int, but that just refused to work in our setup.

[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.Data/Repositories/ProviderCoursesReadRepository.cs]: Ok I reworked to be more vanilla SQL running into a scalar variable

File: src/SFA.DAS.Roatp.Data/Repositories/ProviderCoursesReadRepository.cs [hunk 2]
```diff
@@ -45,13 +48,24 @@ public async Task<List<ProviderCourse>> GetAllProviderCourses(int ukprn)
...
+                    AND PRD.StatusId IN ({OrganisationStatus.Active},{OrganisationStatus.ActiveNotTakingOnApprentices})";
+
+
+            await using var connection = _roatpDataContext.Database.GetDbConnection();
+            await connection.OpenAsync();
+
+            await using var command = connection.CreateCommand();
+            command.CommandText = sql;
+            var result = await command.ExecuteScalarAsync();
+            TryParse(result?.ToString(), out var count);
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Data/Repositories/ProviderCoursesReadRepository.cs:67]: 🤯 Floating TryParse 💥 I would remove the static using... I got stumped when i looked at this 😆

**Review by DevenKShah — APPROVED** [review]

File: src/SFA.DAS.Roatp.Data/Repositories/ProviderCoursesReadRepository.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.Data/Repositories/ProviderCoursesReadRepository.cs]: yeah I didn't leave it floating, looked bad 👃

**Pull request merged**